### DISCUSSION TITLE
Alpine v3 Roadmap

### DISCUSSION THREAD

**Discussion by stephenoldham** [alpine-collective/alpinejs-devtools#239] [Remnant]
So then...just finished catching up on AlpineJs Day stream watching (props @HugoDF on your talk 🙌🏼). What are peoples thoughts re updates to tools and a little UI tweakage in line with the new branding?

**Comment by HugoDF** [comment]
I had a quick look at how to implement the current features on V3 it all looked feasible

**Reply by ryangjchandler** [reply]
I definitely think we should add support for stores.

**Reply by stephenoldham** [reply]
Nice. Yeah, stores would be a great addition for sure. Worth catching peoples use of Spruce for those on v2 as well.

**Reply by ryangjchandler** [reply]
Yeah, depends on what route we take with the DevTools, whether we keep supporting 2.x or start a new major version with support for just 3.x.

**Reply by stephenoldham** [reply]
Makes sense. I guess there may be a fair few people who aren’t able to switch straight away but the upgrade path isn’t major so.

**Reply by HugoDF** [reply]
IE11 support 👀 

Honestly it won't cost us much to keep v2 support we've already got a concept of "try this stuff if the version is above x"

For stores we can support Spruce and v3 stores 👍

**Reply by stephenoldham** [reply]
I mean if it’s not a big performance cost and fairly easy to maintain then I reckon it’s worth it. Would be interesting to see how quickly people upgrade.

**Reply by stephenoldham** [reply]
Ha! just realised I breezed past @HugoDF's need for IE11 😉

**Reply by HugoDF** [reply]
Ah, I've just realised what everyone might've been talking about, there's 2 sides to Alpine v3:

1. supporting v3 sites in devtools (which is captured in https://github.com/alpine-collective/alpinejs-devtools/issues/244#issuecomment-863105862)
2. using v3 to improve the devtools codebase (eg. using Store(s) instead of editing `__x.$data` in https://github.com/alpine-collective/alpinejs-devtools/blob/master/packages/shell-chrome/src/devtools/state.js)

I've pretty much been focusing on 1. but we might need to do part of 2. to keep the devtools working with v3 installed.
